src/Plugins/SimplnxCore/src/SimplnxCore/Filters/IdentifyDuplicateVerticesFilter.hpp

```diff
@@ -0,0 +1,116 @@
+namespace nx::core
+{
+/**
+ * @class IdentifyDuplicateVerticesFilter
+ * @brief This filter will ....
+ */
+class SIMPLNXCORE_EXPORT IdentifyDuplicateVerticesFilter : public IFilter
+{
+public:
+  IdentifyDuplicateVerticesFilter() = default;
+  ~IdentifyDuplicateVerticesFilter() noexcept override = default;
+
+  IdentifyDuplicateVerticesFilter(const IdentifyDuplicateVerticesFilter&) = delete;
+  IdentifyDuplicateVerticesFilter(IdentifyDuplicateVerticesFilter&&) noexcept = delete;
+
+  IdentifyDuplicateVerticesFilter& operator=(const IdentifyDuplicateVerticesFilter&) = delete;
+  IdentifyDuplicateVerticesFilter& operator=(IdentifyDuplicateVerticesFilter&&) noexcept = delete;
+
+  // Parameter Keys
+  static inline constexpr StringLiteral k_InputGeomPath_Key = "input_geom_path";
+  static inline constexpr StringLiteral k_DuplicateMaskPath_Key = "duplicate_mask_path";
+
+  /**
+   * @brief Reads SIMPL json and converts it simplnx Arguments.
+   * @param json
+   * @return Result<Arguments>
+   */
+  static Result<Arguments> FromSIMPLJson(const nlohmann::json& json);
+
+  /**
+   * @brief Returns the name of the filter.
+   * @return
+   */
+  std::string name() const override;
+
+  /**
+   * @brief Returns the C++ classname of this filter.
+   * @return
+   */
+  std::string className() const override;
+
+  /**
+   * @brief Returns the uuid of the filter.
+   * @return
+   */
+  Uuid uuid() const override;
+
+  /**
+   * @brief Returns the human-readable name of the filter.
+   * @return
+   */
+  std::string humanName() const override;
+
+  /**
+   * @brief Returns the default tags for this filter.
+   * @return
+   */
+  std::vector<std::string> defaultTags() const override;
+
+  /**
+   * @brief Returns the parameters of the filter (i.e. its inputs)
+   * @return
+   */
+  Parameters parameters() const override;
+
+  /**
+   * @brief Returns parameters version integer.
+   * Initial version should always be 1.
+   * Should be incremented everytime the parameters change.
+   * @return VersionType
+   */
+  VersionType parametersVersion() const override;
+
+  /**
+   * @brief Returns a copy of the filter.
+   * @return
+   */
+  UniquePointer clone() const override;
+
+protected:
+  /**
+   * @brief Takes in a DataStructure and checks that the filter can be run on it with the given arguments.
+   * Returns any warnings/errors. Also returns the changes that would be applied to the DataStructure.
+   * Some parts of the actions may not be completely filled out if all the required information is not available at preflight time.
+   * @param dataStructure The input DataStructure instance
+   * @param filterArgs These are the input values for each parameter that is required for the filter
+   * @param messageHandler The MessageHandler object
+   * @param shouldCancel Atomic boolean value that can be checked to cancel the filter
+   * @return Returns a Result object with error or warning values if any of those occurred during execution of this function
```

Comment:
doc block missing a parameter description. Look at other filters, there should all be the same